### DISCUSSION TITLE
"too many connections for role"

### DISCUSSION THREAD

**Discussion by andka907** [Budibase/budibase#12229] [Help]
Hi
I am just start to exploring the Budibase and consider myself as beginner when it comes to databases so please forgive me in advance due to my lack of knowledge.

I setup a simple database on elephantsql (free Cloud version).

The database has four tables whereas relationships: 
Table1 1:N to Table 2, Table2 1:N to Table 3, Table3 1:N to Table 4 and Table1 1:N to Table4.
(Table2 have ~15 columns, All tables have less than 5 rows )
Before setting up the relationships I used four screens with Table Blocks of resp table. 
Trying it out in preview it worked perfectly fine. I also confirmed that the relationshsips are working, I seem to have PK and FK setup properly.

After setting the relationsship up - the tables are connected as expected. However when viewing I got "Too many connections for role "XXXXX" " when previewing the application and switching between the views.

In elephantsql there is no any other connections, I explored in pgadmin and could not find any issues. (If I checked correclty).

Now I am not experienced in the console logs in Edge or Firefox and what to look for but I got:
[Client] HTTP 500 on POST:https://caab.budibase.app/api/datasource_plus_c2c71516b84144a7be03f4e09fc95d2d__asset/search
	too many connections for role "XXXXX"
and
{
    "message": "too many connections for role \"XXXXXX\"",
    "status": 500,
    "error": {
        "code": "53300"
    }
}

I tried to find similar issues in this discussion and tried different solutions but not sure if there is any relevant. 

Thank you in advance for advise and help!

**Comment by melohagan** [comment]
Hey @andka907 

The `too many connections for role` error is coming from Postgres. You could have a look at StackOverflow for some ideas: https://stackoverflow.com/questions/25058649/psql-fatal-too-many-connections-for-role

For example one suggestion was: `ALTER ROLE your_username CONNECTION LIMIT -1;`
